### PR TITLE
Add logic path to disable Managed Identity

### PR DESCRIPTION
**Pull request by VariableDeclared**
Managed Identity requires a "Owner" service principal to create and manage role assignments for the managed identities (VMs). This presents the problem of creating a service principal with such permissions.

Creating such application credentials present a security risk as if a malicious user were to gain access they could create accounts/create roles/assign roles/destroy the subscription (Virtual Datacentre) on azure.

This change mitigates this issue by enabling Charmed Kubernetes to use the service principal directly, instead of requiring a service principal with board permissions to assign roles.

Related [Azure integrator charm change](https://github.com/juju-solutions/charm-azure-integrator/pull/30)
Related [Interface change](https://github.com/juju-solutions/interface-azure-integration/pull/6)

Fixes [LP#1928906](https://bugs.launchpad.net/charm-azure-integrator/+bug/1928906)

- [x] Test disable managed identity path
- [x]  Test "credentials" file path
- [x]  Test juju grant path

### CODE REVIEW

**Review by Cynerva — CHANGES_REQUESTED** [review]
This branch conflicts with master. Please resolve conflicts.

**Comment by VariableDeclared** [comment]
Fixed! @Cynerva

File: lib/charms/layer/kubernetes_common.py [hunk 1]
```diff
@@ -604,6 +604,11 @@ def write_azure_snap_config(component):
                 "subnetName": azure.subnet_name,
                 "securityGroupName": azure.security_group_name,
                 "loadBalancerSku": "standard",
+                "securityGroupResourceGroup": azure.security_group_resource_group,
+                "loadBalancerSku": "standard",
```
Comment (johnsca) [inline lib/charms/layer/kubernetes_common.py]: The `loadBalancerSku` field got duplicated.

```suggestion
                "securityGroupResourceGroup": azure.security_group_resource_group,
                "loadBalancerSku": "standard",
```

**Review by johnsca — CHANGES_REQUESTED** [review]

File: lib/charms/layer/kubernetes_common.py (hunk 1 above)
Comment (VariableDeclared) [reply lib/charms/layer/kubernetes_common.py]: Removed the extra loadBalancerSku key

thanks!

**Review by johnsca — APPROVED** [review]

**Comment by VariableDeclared** [comment]
@johnsca linting blocked the merge, fixed the offending file

**Pull request merged**